src/main/java/org/springframework/samples/petclinic/owner/Owner.java

```diff
@@ -95,9 +95,17 @@ public List<Pet> getPets() {
 
 	public void addPet(Pet pet) {
 		if (pet.isNew()) {
-			getPets().add(pet);
+		    getPets().add(pet);
+		} else {
+		    for (int i = 0; i < getPets().size(); i++) {
+		        if (getPets().get(i).getId().equals(pet.getId())) {
+		            getPets().set(i, pet);
+		            return;
```

Comment:
I might set a flag like `petAdded = true` here rather than returning, because if there is other logic in the function that would need to be performed, it would be skipped by the return. As a code style thing, `return` in the middle of the function is usually not preferred. It won't break this code as it is now, it's more of a consideration for future modifications.

Reply:
Here's a reference: https://stackoverflow.com/a/733858

Reply:
I think that the return is intentional; without the return a duplicate would be added in line 106 below.